CreateSymLinkFolder.ps1

```diff
@@ -1,17 +1,203 @@
+        private readonly static MethodInfo s_createVistaDialogMethodInfo = typeof(OpenFileDialog).GetMethod("CreateVistaDialog", c_flags);
+        private readonly static MethodInfo s_onBeforeVistaDialogMethodInfo = typeof(OpenFileDialog).GetMethod("OnBeforeVistaDialog", c_flags);
+        private readonly static MethodInfo s_getOptionsMethodInfo = typeof(FileDialog).GetMethod("GetOptions", c_flags);
+        private readonly static MethodInfo s_setOptionsMethodInfo = s_iFileDialogType.GetMethod("SetOptions", c_flags);
+        private readonly static uint s_fosPickFoldersBitFlag = (uint) s_windowsFormsAssembly
+            .GetType("System.Windows.Forms.FileDialogNative+FOS")
+            .GetField("FOS_PICKFOLDERS")
+            .GetValue(null);
+        private readonly static ConstructorInfo s_vistaDialogEventsConstructorInfo = s_windowsFormsAssembly
+            .GetType("System.Windows.Forms.FileDialog+VistaDialogEvents")
+            .GetConstructor(c_flags, null, new[] { typeof(FileDialog) }, null);
+        private readonly static MethodInfo s_adviseMethodInfo = s_iFileDialogType.GetMethod("Advise");
+        private readonly static MethodInfo s_unAdviseMethodInfo = s_iFileDialogType.GetMethod("Unadvise");
+        private readonly static MethodInfo s_showMethodInfo = s_iFileDialogType.GetMethod("Show");
+
+        public static ShowDialogResult Show(IntPtr ownerHandle, string initialDirectory, string title, string description) {
+            var openFileDialog = new OpenFileDialog {
+                AddExtension = false,
+                CheckFileExists = false,
+                DereferenceLinks = true,
+                Filter = c_foldersFilter,
+                InitialDirectory = initialDirectory,
+                Multiselect = false,
+                Title = title
+            };
+
+            var iFileDialog = s_createVistaDialogMethodInfo.Invoke(openFileDialog, new object[] { });
+            s_onBeforeVistaDialogMethodInfo.Invoke(openFileDialog, new[] { iFileDialog });
+            s_setOptionsMethodInfo.Invoke(iFileDialog, new object[] { (uint) s_getOptionsMethodInfo.Invoke(openFileDialog, new object[] { }) | s_fosPickFoldersBitFlag });
+            var adviseParametersWithOutputConnectionToken = new[] { s_vistaDialogEventsConstructorInfo.Invoke(new object[] { openFileDialog }), 0U };
+            s_adviseMethodInfo.Invoke(iFileDialog, adviseParametersWithOutputConnectionToken);
+
+            try {
+                int retVal = (int) s_showMethodInfo.Invoke(iFileDialog, new object[] { ownerHandle });
+                return new ShowDialogResult {
+                    Result = retVal == 0,
+                    FileName = openFileDialog.FileName
+                };
+            }
+            finally {
+                s_unAdviseMethodInfo.Invoke(iFileDialog, new[] { adviseParametersWithOutputConnectionToken[1] });
+            }
+        }
+    }
```

Comment:
what the fuck. look, it's even using reflection to get names directly.
why is this necessary again? Windows XP nor windows Vista can't even run powershell scripts....